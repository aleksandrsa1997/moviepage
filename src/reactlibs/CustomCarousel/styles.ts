export const responsive = {
    desktop: {
        breakpoint: { max: 3000, min: 1024 },
        items: 4,
        slidesToSlide: 4 // optional, default to 1.
    }
};