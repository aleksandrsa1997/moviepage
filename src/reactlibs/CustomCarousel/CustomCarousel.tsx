import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';
import React from 'react';
import { IMovie } from 'src/store/movie/types';
import FilmCard from 'src/reactlibs/FilmCard';
import { IMAGE_URL } from 'src/constants/api';
import { responsive } from './styles';
import Link from 'next/link';

export type TProps = {
    movies: Array<IMovie>,
    sortedMovies: any,
    setIdToStore: (value: string) => void;
    getMovieInfoById: () => void;
}

const setting = {
    ssr: true, // means to render carousel on server-side.
    showDots: true,
    infinite: true,
    swipeable: false,
    draggable: false,
    itemClass: "carousel-item-padding-40-px",
    responsive: responsive,
    dotListClass: "custom-dot-list-style",
    autoPlaySpeed: 1000,
    containerClass: "carousel-container",
    keyBoardControl: true,
    customTransition: "all .5",
    transitionDuration: 500,
    removeArrowOnDeviceType: ["tablet", "mobile"],
}
// @ts-ignore
const CustomCarousel = ({ sortedMovies, setIdToStore, getMovieInfoById }: TProps) => {
    const handleClick = (event: React.MouseEvent<HTMLButtonElement, MouseEvent>) => {
        // setIdToStore((event.target as HTMLButtonElement).id);
        console.log((event.target as HTMLButtonElement).id);
        // @ts-ignore
        getMovieInfoById((event.target as HTMLButtonElement).id);
    }

    return (<Carousel { ...setting } >
        {sortedMovies.map((item: any) =>
        <Link href='/about_movie'>
            <FilmCard
                id={item.id}
                key={ item.title }
                alt='Movie Poster'
                color='textSecondary'
                title={ item.title }
                image={`${IMAGE_URL}${item.poster_path}`}
                variant='body2'
                overview={item.overview}
                popularity={ item.popularity }
                vote_count={ item.vote_count }
                vote_average={ item.vote_average }
                original_title={item.original_title}
                onClick={handleClick}
            />
            </Link>
        )}
    </Carousel>
    );
}
export default CustomCarousel;