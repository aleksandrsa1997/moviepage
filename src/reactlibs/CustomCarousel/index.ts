import CustomCarousel from './CustomCarousel';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';
import * as selectors from 'src/store/movie/selectors';
import { IMovies } from 'src/store/movie/reducer';
import * as actions from 'src/store/movie/action';

const mapStateToProps = (state: IMovies) => ({
    movies: selectors.getMovies(state),
    sortedMovies: selectors.sortedMoviesSelector(state),
});

const mapDispatchToProps = (dispatch: Dispatch) => ({
    getMovieInfoById: (payload: string) => dispatch(actions.getMovieInfoById(payload)),
    setIdToStore: (payload: string) => dispatch(actions.setIdToStore(payload)),
});
// @ts-ignore
export default connect(mapStateToProps, mapDispatchToProps)(CustomCarousel);