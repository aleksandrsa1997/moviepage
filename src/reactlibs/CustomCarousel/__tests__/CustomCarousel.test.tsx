import React from 'react';
import CustomCarousel from '../index';
import configureStore from 'redux-mock-store';

describe('CustomCarousel snapshot', () => {
    let store;

    beforeEach(()=> {
        store = mockStore({
            movies: [
                {
                    adult: false,
                    backdrop_path: '/7KL4yJ4JsbtS1BNRilUApLvMnc5.jpg',
                    genre_ids: [18, 53],
                    id: 649087,
                    original_language: 'sv',
                    original_title: 'Red Dot',
                    overview: 'On a hiking trip to rekindle their marriage',
                    popularity: 1040.699,
                    poster_path: '/xZ2KER2gOHbuHP2GJoODuXDSZCb.jpg',
                    release_date: '2021-02-11',
                    title: 'Red Dot',
                    video: false,
                    vote_average: 6.3,
                    vote_count: 337
                }
            ]
        });
    })

    it('should component renderer correctly', () => {
        const component = mountSmart(<CustomCarousel/>, store);

        expect(component).toMatchSnapshot();
    });

    it('should component renderer correctly CustomCarousel', () => {
        const component = mountSmart(<CustomCarousel/>, store);

        expect(component.find('CustomCarousel')).toHaveLength(1);
    });
});