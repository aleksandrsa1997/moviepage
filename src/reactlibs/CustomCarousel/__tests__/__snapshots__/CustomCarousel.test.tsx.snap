// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomCarousel snapshot should component renderer correctly 1`] = `ReactWrapper {}`;
